src/langcheck/eval/_validation.py

```diff
@@ -0,0 +1,145 @@
+from __future__ import annotations
+
+from typing import List, Optional
+
+
+def validate_parameters_reference_based(
+        generated_outputs: List[str] | str,
```

Comment:
What do you think about add name to `List[str] | str` and use that as the type annotation in each function?
I think it would be easier to read, and also make it easier to add more syntax sugar in the future.

Reply:
`List[str] | str` isn't too long, and I think it's more clear for users reading the documentation who aren't familiar with custom types in our codebase.

Reply:
Sounds reasonable, let's leave it as is then.